src/components/Tiptap.tsx

```diff
@@ -0,0 +1,102 @@
+import { forwardRef, useImperativeHandle } from 'react';
+import { BubbleMenu, EditorContent, useEditor } from '@tiptap/react';
+import HardBreak from '@tiptap/extension-hard-break';
+import StarterKit from '@tiptap/starter-kit';
+import Placeholder from '@tiptap/extension-placeholder';
+import {
+    RiBold,
+    RiItalic,
+    RiStrikethrough,
+    RiListUnordered,
+} from 'react-icons/ri';
+import ToggleOptions from './ToggleOption';
+
+type Options = {
+    bold: any;
+    italic: any;
+    strike: any;
+    bulletlist: any;
+};
+
+const Tiptap = forwardRef(({ onFieldChange }: any, ref: any) => {
+    const LineBreak = HardBreak.extend({
+        addKeyboardShortcuts() {
+            return {
+                Enter: () => this.editor.commands.setHardBreak(),
+            };
+        },
+    });
+    const editor: any = useEditor({
+        extensions: [
+            StarterKit.configure({
+                hardBreak: false,
+            }),
+            LineBreak,
+            Placeholder.configure({
+                placeholder: 'Descrição da vaga',
+                emptyEditorClass:
+                    'cursor-text before:content-[attr(data-placeholder)] before:absolute before:top-3 before:left-4 before:text-mauve-11 before:opacity-30 before-pointer-events-none',
+            }),
+        ],
+        editorProps: {
+            attributes: {
+                class: 'w-full overflow-auto h-32 border py-3 px-4 mt-[10px] rounded-md block outline-none focus:border-blue',
+            },
+        },
+
+        content: ``,
+        onUpdate: ({ editor }) => {
+            const html = editor.getHTML();
+            onFieldChange(html);
+        },
+    });
+
+    useImperativeHandle(ref, () => ({
+        clearContent() {
+            editor.commands.clearContent();
+        },
+    }));
+    const ToggleOption = ['bold', 'italic', 'strike', 'bulletlist'];
+    const ToggleFunctions: Options = {
```

Comment:
Essa parte poderia alterar os nomes. Seguindo a convenção de clean code para nomeação de constantes 
```suggestion
    const TOGGLE_OPTIONS = ['bold', 'italic', 'strike', 'bulletlist'];
    const TOGGLE_FUNCTIONS: Options = {
```